internal/workflow_testsuite.go

```diff
@@ -306,6 +307,17 @@ func (t *TestWorkflowEnvironment) OnRequestCancelExternalWorkflow(domainName, wo
 	return t.wrapCall(call)
 }
 
+// OnGetVersion setup a mock for workflow.GetVersion() call. By default, if mock is not setup, the GetVersion call from
+// workflow code will always return the maxSupported version. Make it not possible to test old version branch. With this
+// mock support, it is possible to test code branch for different versions.
+// Note: if mock for GetVersion is setup, the test framework expect all calls to GetVersion from the tested workflow are
+// all mocked. It is not supported to have one GetVersion call mocked and the second GetVersion call not mocked. You need
+// to either not mock any GetVersion call, or mock all GetVersion calls.
```

Comment:
I don't think it is acceptable. The problem is that the workflow might reference some library that calls GetVersion but the implementer of the workflow might not even know about it. I believe it should be possible to mock each "changeId" individually.

Reply:
good point, let me update